couchdb/tests/couch-cluster.spec.js

```
@@ -0,0 +1,125 @@
+const chai = require('chai');
+const { expect } = chai;
+
+
+describe('CouchDB Cluster Unit Tests', () => {
+  const COUCHDB_USER = process.env.COUCHDB_USER;
+  const COUCHDB_PASSWORD = process.env.COUCHDB_PASSWORD;
+  const COUCH_SERVERS = ['couchdb1.local', 'couchdb2.local', 'couchdb3.local'];
+  const TEST_DB_NAME = 'tests';
+
+  const requestOnNode = async (serverIndex, path, options = {}) => {
+    const server = COUCH_SERVERS[serverIndex];
+    const url = `http://${server}:5984${path}`;
+    
+    const fetchOptions = {
+      method: options.method || 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Authorization': `Basic ${Buffer.from(`${COUCHDB_USER}:${COUCHDB_PASSWORD}`).toString('base64')}`,
+        ...options.headers
+      }
+    };
+
+    if (options.body) {
+      fetchOptions.body = JSON.stringify(options.body);
+    }
+    
+    try {
+      const response = await fetch(url, fetchOptions);
+      const body = await response.json().catch(() => ({}));
```

I think that we should throw an error when the `.json` call fails.